src/main/java/frc/robot/poseestimator/GBPoseEstimator.java

```diff
@@ -179,8 +182,12 @@ private boolean isObservationTooOld(VisionObservation visionObservation) {
 	private void addVisionObservation(VisionObservation observation) {
 		Optional<Pose2d> odometryInterpolatedPoseSample = odometryPoseInterpolator.getSample(observation.timestamp());
 		odometryInterpolatedPoseSample.ifPresent(odometryPoseSample -> {
+			visionDenoiser.addVisionObservation(observation);
+			VisionObservation fixedObservation;
+			Optional<VisionObservation> fixedOptionalObservation = visionDenoiser.calculateLinearFilterResult();
+			fixedObservation = fixedOptionalObservation.orElse(observation);
```

Comment:
are you sure that returning the observation if not present is ok?

Reply:
the worst thing that could happen is a bit of vibrations. Also, this happens iff the inner LinkedList of observations is empty, which happens just for the first observations (we can't reduce the vibrations in that state since we don't have any addtional data).

Reply:
right